### ISSUE TITLE
Dependency wrapper for Ella plugins

### ISSUE THREAD

**Issue by xaralis** [ella/ella#56]
It would be nice to create some dependency wrapper between plugins.  Nice example can be **imports** app. This one needs **articles**, but one can imagine articles can very nicely live without **imports** so it doesn't make sense to merge the apps.

Instead, we should create some dependency-checking mechanism for each module. Putting the dependency into **setup.py** is not sufficent, because the fact that it's installed doesn't necessarily mean that it's included in `INSTALLED_APPS`  too.

The wrapper can be fairly simple, just simple checking and in case that dependencies are not met, warning should be raised.

What do you think?

**Comment by whit** [comment]
+1. In some cases we need check ordering of `INSTALLED_APPS` too. E.g. [ella-cateory-subdomain](https://github.com/SanomaCZ/ella-category-subdomain) must be first.

**Comment by whit** [comment]
@HonzaKral Is something similar planned in Django (maybe with `INSTALLED_APPS` refactoring)?

**Comment by honzakral** [comment]
I really don't think anything like this is needed. Documentation should be more than sufficient to get things running. Ella's target audience is still developers, not designers or end-users.

You are never getting it both easy and flexible enough to do everything right. Just use requirements.txt and REAMDE, should be more than enough.

**Comment by honzakral** [comment]
Also for https://github.com/SanomaCZ/ella-category-subdomain - just PLEASE use https://docs.djangoproject.com/en/dev/ref/settings/#absolute-url-overrides and don't monkeypatch stuff. You should only need to change the method on Publishable.

We might need this too (we think about having custom URLs file) so I might even provide some hook into ella itself.

**Comment by whit** [comment]
@HonzaKral I didn't know. But it's usable to mapping only selected categories?

**Comment by xaralis** [comment]
I don't think that we should limit our users to developers only. An intermediate html coder with some basic knowledge of Django templating should be able to run Ella with proper support in the docs too. This is actually quite common scenario (html coders running the project) and definitely something we shouldn't reject.

I think, that adding some basic (and I mean really basic, simple try: import xyz, do some basic check,... except ImportError: raise ... should be right enough) dependency checking to the Ella plugins would help a lot and make the code much more plugin-like. 

Yes, you're right that for developers, this is probably unneeded. But for the others info message saying "you have to install app XYZ before using this module" is far more useful than "ImportError: Cannot import app ZYX".

Don't you think?

**Comment by xaralis** [comment]
It seems that this is simply a design decision after all: are we creating Ella for close group of (interested) people who know what is required or do we want to make it dumb-user-friendly to ease the use for people that didn't have a chance to run into it yet?

Don't take it too seriously, it's not meant this way. But it's a basic decision we should do.

**Comment by honzakral** [comment]
That is very much unreasonable - Ella is way too low level (which for us is a _very_ good thing) to be usable by non-python people. Having people go through the django tutorial is usually enough for them to grasp the basic ideas so then they can read the docs and use the individual plug-ins.

Ella will most likely never catch up with django-cms, wordpress, drupal or other designer-friendly CMSs, nor should it. Let's be realistic here and not try to cater to a crowd that will never use Ella without a programmer's help.

**Comment by xaralis** [comment]
I don't see a reason why Ella should catch up with django-cms in a matter of developer-friendliness. When there is no need to customize stuff, it's quite simple to get site up and running in matter of hours even for a non-programmer (having some basic knowledge of django templates). Actually, it felt even simpler than Django CMS to me. All you need is understanding of template tags Ella provides and core principles which it is built on. All in all, it's nothing complicated and it seems like a shame to get rid of these potential users.

**Comment by martinbilek** [comment]
Dependency wrapper looks like "nice to have" feature for me. Dependencies should be solved on django-layer...or let's remember this good idea for future ella versions, when we have more time to discuss/think about/implement such functionality.

**Comment by xaralis** [comment]
Let's keep this as nice-to-have then. Truth is that we don't really need this right now.

Removing "cleanup" tag.
